keyvault/client/keys/src/main/java/com/azure/keyvault/keys/KeyClient.java

```diff
@@ -0,0 +1,639 @@
+        KeyImportRequestParameters parameters = new KeyImportRequestParameters().key(keyMaterial);
+        return service.importKey(endpoint, name, API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Imports an externally created key and stores it in key vault. The import key operation may be used to import any key type
+     * into the Azure Key Vault. If the named key already exists, Azure Key Vault creates a new version of the key. This operation requires the {@code keys/import} permission.
+     *
+     * <p>The {@code keyImportOptions} is required and its fields {@link KeyImportOptions#name() name} and {@link KeyImportOptions#keyMaterial() key material} cannot
+     * be null. The {@link KeyImportOptions#expires() expires} and {@link KeyImportOptions#notBefore() notBefore} values in {@code keyImportOptions} are optional. If not
+     * specified, no values are set for the fields. The {@link KeyImportOptions#enabled() enabled} field is set to true and the {@link KeyImportOptions#hsm() hsm} field is
+     * set to false by Azure Key Vault, if they are not specified.</p>
+     *
+     * <p><strong>Code Samples</strong></p>
+     * <p>Imports a new key into key vault. Prints out the details of the imported key.</p>
+     * <pre>
+     * KeyImportOptions keyImportOptions = new KeyImportOptions("keyName", jsonWebKeyToImport)
+     *   .hsm(true)
+     *   .expires(OffsetDateTime.now().plusDays(60));
+     *
+     * Key importedKey = keyClient.importKey(keyImportOptions).value();
+     * System.out.printf("Key is imported with name %s and id %s \n", importedKey.name(), importedKey.id());
+     * </pre>
+     *
+     * @param keyImportOptions The key import configuration object containing information about the json web key being imported.
+     * @throws NullPointerException if {@code keyImportOptions} is {@code null}.
+     * @throws HttpRequestException if {@code name} is empty string.
+     * @return A {@link Response} whose {@link Response#value() value} contains the {@link Key imported key}.
+     */
+    public Response<Key> importKey(KeyImportOptions keyImportOptions) {
+        Objects.requireNonNull(keyImportOptions, "The key import configuration parameter cannot be null.");
+        KeyImportRequestParameters parameters = new KeyImportRequestParameters()
+                .key(keyImportOptions.keyMaterial())
+                .hsm(keyImportOptions.hsm())
+                .keyAttributes(new KeyRequestAttributes(keyImportOptions));
+        return service.importKey(endpoint, keyImportOptions.name(), API_VERSION, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE).block();
+    }
+
+    /**
+     * Gets the public part of the specified key and key version. The get key operation is applicable to all key types and
+     * If the requested key is symmetric, then no key material is released in the response. This operation requires the {@code keys/get} permission.
```

Comment:
>If the requested key is symmetric, then no key material is released in the response. [](start = 7, length = 84)

Currently key vault doesn't support symmetric keys through the key api.

Reply:
excluded this from javadoc.